http-clients/src/main/java/org/triplea/http/client/web/socket/WebSocketConnection.java

```diff
@@ -68,52 +80,28 @@
 
   WebSocketConnection(final URI serverUri) {
     this.serverUri = serverUri;
-    client =
-        new WebSocketClient(serverUri) {
-          @Override
-          public void onOpen(final ServerHandshake serverHandshake) {
-            synchronized (queuedMessages) {
-              connectionIsOpen = true;
-              queuedMessages.forEach(this::send);
-              queuedMessages.clear();
-            }
-          }
-
-          @Override
-          public void onMessage(final String message) {
-            listener.messageReceived(message);
-          }
-
-          @Override
-          public void onClose(final int code, final String reason, final boolean remote) {
-            if (remote) {
-              log.severe("Connection to server closed: " + reason);
-            }
-            pingSender.cancel();
-            listener.connectionClosed(reason);
-          }
-
-          @Override
-          public void onError(final Exception exception) {
-            listener.handleError(exception);
-          }
-        };
     pingSender =
         Timers.fixedRateTimer("websocket-ping-sender")
             .period(45, TimeUnit.SECONDS)
             .delay(45, TimeUnit.SECONDS)
             .task(
                 () -> {
-                  if (client.isOpen()) {
-                    client.sendPing();
+                  if (!client.isOutputClosed()) {
+                    client
+                        .sendPing(ByteBuffer.allocate(0))
+                        .exceptionally(logWebSocketError(Level.INFO, "Failed to send ping."));
                   }
                 });
   }
 
   /** Does an async close of the current websocket connection. */
   void close() {
     closed = true;
-    client.close();
+    if (client != null && !client.isOutputClosed()) {
```

Comment:
Is it very burdensome to keep client non-nullable? Also wondering why it's nullable now.

Reply:
I'm actually not sure about this, but when testing the customHost thingy I got an NPE in this line, presumably because a connection never got opened (maybe concurrency related) and the connection was closed before that ever happened

Reply:
I just tried to debug this issue and it's gone. Seems like the issue really was that the initial connection took too long, so that the connection was closed before it even really opened.

I'll add a comment clarifying that